pkg/config/analysis.go

```diff
@@ -33,6 +37,20 @@ type AnalysisExpected struct {
 	Max *float64 `json:"max"`
 }
 
+// InRange returns true if the given value is within the range.
+func (e *AnalysisExpected) InRange(value float64) (bool, error) {
+	if e.Min == nil && e.Max == nil {
```

Comment:
I think this check should not be done here. Instead, this validation should be done once before using this `AnalysisExpected`.  Basically, the `InRange` function will return only a single bool value.

Reply:
That is so true. Let me do so.